ui/selectors/transactions.js

```diff
@@ -97,18 +97,23 @@ export const unapprovedEncryptionPublicKeyMsgsSelector = (state) =>
 export const unapprovedTypedMessagesSelector = (state) =>
   state.metamask.unapprovedTypedMessages;
 
-export const smartTransactionsListSelector = (state) =>
-  state.metamask.smartTransactionsState?.smartTransactions?.[
+export const smartTransactionsListSelector = (state) => {
+  const { address: selectedAddress } = getSelectedInternalAccount(state);
+  return state.metamask.smartTransactionsState?.smartTransactions?.[
     getCurrentChainId(state)
   ]
-    ?.filter((stx) => !stx.confirmed)
+    ?.filter((stx) => {
+      const { txParams } = stx;
+      return txParams?.from === selectedAddress && !stx.confirmed;
+    })
     .map((stx) => ({
       ...stx,
-      transactionType: TransactionType.smart,
+      isSmartTransaction: true,
```

Comment:
Do we need this since we're creating all the new smart transactions through the `TransactionController` meaning they will already be visible via the standard transaction controller selectors?

Reply:
I believe so, because it's used in the transaction-list.component.js file. With doing `transactionType: TransactionType.smart` we were losing the original `transactionType` value, which we don't want.